Fix request validation

din-mukhammed opened the pull request:
### Changes
went through all endpoints and fixed minor error checks
recovery is already in place:
https://github.com/0chain/blobber/blob/9c0bac6f454838ca77948f15d91ef5a80176f64f/code/go/0chain.net/blobbercore/handler/handler.go#L152
https://github.com/0chain/blobber/blob/9c0bac6f454838ca77948f15d91ef5a80176f64f/code/go/0chain.net/validatorcore/storage/handler_main.go#L15

### Fixes



### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/blobber/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- 0chain:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

din-mukhammed committed:
fix error handling

din-mukhammed committed:
Merge remote-tracking branch 'origin/staging' into 1048-revisit-request-validation

din-mukhammed committed:
fix st

din-mukhammed committed:
fix typo

din-mukhammed committed:
fix ut

dabasov commented:
https://github.com/0chain/blobber/blob/staging/code/go/0chain.net/validatorcore/storage/challenge_handler.go

please also check this file, it is validator endpoints

din-mukhammed committed:
Merge branch 'staging' into 1048-revisit-request-validation

din-mukhammed committed:
Merge branch 'staging' into 1048-revisit-request-validation

din-mukhammed committed:
add marker string fields len check

din-mukhammed commented:
> https://github.com/0chain/blobber/blob/staging/code/go/0chain.net/validatorcore/storage/challenge_handler.go
> 
> please also check this file, it is validator endpoints

recovery is also implemented in validator endpoints,
https://github.com/0chain/blobber/blob/f3329e7509bb5a3be4d117a47b389647f76ded0e/code/go/0chain.net/validatorcore/storage/handler_main.go#L15
 other validations seem to be sufficient. 

On a separate note, added readmarker and writemarker string fields' len check
